contracts/masset/shared/MassetHelpers.sol

```diff
@@ -18,25 +18,20 @@ library MassetHelpers {
     using SafeMath for uint256;
     using SafeERC20 for IERC20;
 
-    function transferTokens(
+
+    function transferReturnBalance(
         address _sender,
         address _recipient,
         address _basset,
-        bool _erc20TransferFeeCharged,
         uint256 _qty
     )
         internal
-        returns (uint256 receivedQty)
+        returns (uint256 receivedQty, uint256 recipientBalance)
     {
-        receivedQty = _qty;
-        if(_erc20TransferFeeCharged) {
-            uint256 balBefore = IERC20(_basset).balanceOf(_recipient);
-            IERC20(_basset).safeTransferFrom(_sender, _recipient, _qty);
-            uint256 balAfter = IERC20(_basset).balanceOf(_recipient);
-            receivedQty = StableMath.min(_qty, balAfter.sub(balBefore));
-        } else {
-            IERC20(_basset).safeTransferFrom(_sender, _recipient, _qty);
-        }
+        uint256 balBefore = IERC20(_basset).balanceOf(_recipient);
+        IERC20(_basset).safeTransferFrom(_sender, _recipient, _qty);
+        recipientBalance = IERC20(_basset).balanceOf(_recipient);
+        receivedQty = StableMath.min(_qty, recipientBalance.sub(balBefore));
```

Comment:
I might be worth considering that this function may fail if `recipientBalance < balBefore`. I suppose this can, for instance, happen if _sender and _recipient are identical and there is a transfer fee.

Reply:
Given that we trust the caller here - I see no reason why one caller would ever want to transfer tokens to itself and no reason to account for this scenario

Reply:
Yeah, you can always look at the callers to make sure this doesn't happen. For me, it simplifies reasoning about library functions if they make as few assumptions as possible. However, you're right that for its current callers this shouldn't be an issue.